src/main/java/de/tum/cit/aet/artemis/communication/repository/MessageSpecs.java

```diff
@@ -42,12 +43,15 @@ public static Specification<Post> getSearchTextAndAuthorSpecification(String sea
         return (root, query, criteriaBuilder) -> {
             boolean hasText = searchText != null && !searchText.isBlank();
             boolean hasAuthors = authorIds != null && authorIds.length > 0;
+            // no author ids and no search text means no filtering, return all posts
```

Comment:
To me "return all posts" reads a little odd within this method, considering that we return null here (which means not filtering)

```suggestion
            // no author ids and no search text means no filtering
```